### PR TITLE
Nav data sampled based on layout width and number of data points.

### PR DESCRIPTION
**Pull request by jdhodges**
#674.

### CODE REVIEW

**Comment by jdhodges** [comment]
Fixed some indentation so it's a good idea to omit white space changes
https://github.com/ScottLogic/BitFlux/pull/679/files?w=1

File: src/assets/js/chart/nav.js [hunk 1]
```diff
@@ -130,20 +134,21 @@ export default function() {
     function nav(selection) {
         var model = selection.datum();
 
-        createDefs(selection, model.data);
+        sampler.bucketSize((model.data.length * layoutWidth) / 1e5);
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Why `1e5`? Feels like a magic number?

Would this be better expressed as a function where we can pass in a desired number of buckets and a width to generate an appropriate bucket size?

```js
// for example: bucketSizeForBucketsPerWidth(1, 10)
// could return a bucketSize which will give us 1 bucket every 10 pixels

```

File: src/assets/js/chart/nav.js [hunk 2]
```diff
@@ -19,82 +19,88 @@ export default function() {
...
+                    [viewScale.domain()[0], navChart.yDomain()[0]],
+                    [viewScale.domain()[1], navChart.yDomain()[1]]
+                ]);
+            } else {
+                // This stops the brush data being overwritten by the point data
+                return this.data;
+            }
+        });
 
+    var maskXScale = fc.scale.dateTime();
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Is this unused now?

File: src/assets/js/chart/nav.js [hunk 3]
```diff
@@ -19,82 +19,88 @@ export default function() {
...
     var maskYScale = d3.scale.linear();
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Is this unused now?

File: src/assets/js/chart/nav.js [hunk 4]
```diff
@@ -120,35 +126,41 @@ export default function() {
...
 
+    function bucketSizeForBucketsPerWidth(buckets, width, numberOfPoints) {
+        var bucketSize = numberOfPoints / ((buckets / width) * layoutWidth);
+        return bucketSize > 1 ? bucketSize : 1;
+    }
+
     function nav(selection) {
         var model = selection.datum();
 
+        sampler.bucketSize(bucketSizeForBucketsPerWidth(1, 10, model.data.length));
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: I know this was my suggestion - but after trying it out, I'm not convinced that varying the bucket size based on width is such a good idea - sorry!

I think adjusting it based upon the number of data points is still a good idea though.

Let me know if you disagree :)

File: src/assets/js/chart/nav.js [hunk 5]
```diff
@@ -166,25 +178,26 @@ export default function() {
                     dispatch[event.viewChange](util.domain.centerOnDate(
                         model.discontinuityProvider,
                         viewScale.domain(),
-                        model.data,
+                        sampledData,
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Should we use sampled data or data here?

File: src/assets/js/chart/nav.js [hunk 6]
```diff
@@ -166,25 +178,26 @@ export default function() {
...
+            .scale(viewScale)
+            .trackingLatest(model.trackingLatest)
+            .allowPan(false)
+            .on('zoom', function(domain) {
+                dispatch[event.viewChange](domain);
+            });
 
         selection.select('.plot-area')
-          .call(zoom);
+            .datum({data: sampledData})
```
Comment (jleft) [inline src/assets/js/chart/nav.js]: Should we use sampled data or data here?

**Pull request closed without merging**